File: code/backend/DarkDeeds.TaskServiceApp.Models/Dto/PlannedRecurrenceDto.cs
﻿using System;
using DarkDeeds.TaskServiceApp.Entities.Enums;

namespace DarkDeeds.TaskServiceApp.Models.Dto
{
    public class PlannedRecurrenceDto
    {
        public string Uid { get; set; }
        
        public string Task { get; set; }
        
        public DateTime StartDate { get; set; }
        
        public DateTime? EndDate { get; set; }
        
        public int? EveryNthDay { get; set; }
        
        public string EveryMonthDay { get; set; }

        public RecurrenceWeekdayEnum? EveryWeekday { get; set; }
        
        public bool IsDeleted { get; set; }
    }
}

File: code/frontend4/src/recurrences/models/PlannedRecurrenceDto.ts
import { RecurrenceWeekdayEnum } from './RecurrenceWeekdayEnum'

export class PlannedRecurrenceDto {
    constructor(
        public uid: string,
        public task: string,
        public startDate: Date,
        public endDate: Date | null,
        public everyNthDay: number | null,
        public everyMonthDay: string | null,
        public everyWeekday: RecurrenceWeekdayEnum | null,
        public isDeleted: boolean,
    ) {}
}
